docs/examples/2Dwavefunc_animated.py

```diff
@@ -0,0 +1,184 @@
+"""
+===============================================
+Animating a time-varying 2D wave function
+===============================================
+
+This is a simple demonstration of how one can visualize
+time-varying 2D wave functions using FURY.
+
+Some other interesting functions you can try plotting -
+#z = (x**2 - y**2)/(x**2 + y**2)**0.5*np.cos(np.pi*x-rate*time)
+#z = np.sin(x**2)*np.cos(y**2-rate*incre)#*np.sin(rate*time)
+#z = np.sin(x**2-y**2)*np.sin(time)
+
+"""
+
+###############################################################################
+# Importing necessary modules
+
+from fury import window, actor, ui, utils, colormap
+import numpy as np
+import itertools
+
+
+###############################################################################
+# The following function is used to create and/or update the coordinates of the
+# points which are being used to plot the wave.
+# Kindly note that only the z coordinate is being modified with time as only
+# the z coordinate is a function of time.
+
+def update_coordinates(time=0, incre=0, cmap_name='viridis'):
+
+    # x, y points which will be used to fit the equation to get elevation and
+    # generate the surface.
+    global x, y
+
+    # manipulate rate to change the speed of the waves; (default = 1)
+    rate = 1
+
+    # Z is the function F i.e. F(x, y, t)
+    z = 0.48*np.sin(2*np.pi*x)*np.cos(2*np.pi*y-rate*time)
```

Comment:
If multiple functions are shown maybe x, y and z could be an input of the function.